src/main/java/Duke.java

```diff
@@ -1,10 +1,190 @@
+package main.java;
+
+import java.util.Scanner;
+import java.util.regex.Pattern;
+
 public class Duke {
+
+    /* Variables for Duke's use */
+    private static Task[] taskList;
+    private static int listIndex;
+    private static Scanner scanner;
+
+    /* Constants */
+    private static String INDENT = "\n    ";
+    private static String H_LINE = INDENT +
+            "------------------------------------------------";
+
+
     public static void main(String[] args) {
+
+        //Initialize necessary variables
+        init();
+
+        //Print introduction to Duke
+        introduction();
+
+        //Re-iterate what the user types, store in list, and unmark / mark
+        while (!respondToUser()) {
+        }
+
+        //If the user exits, salute them goodbye
+        goodbye();
+
+    }
+
+    /* Initialize variables for Duke's use */
+    private static void init() {
+        scanner = new Scanner(System.in);
+        taskList = new Task[100];
+        listIndex = 0;
+    }
+
+    /* Print introduction to Duke */
+    private static void introduction() {
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        String introText = "\n    Hello! I'm Duke\n    What can I do for you?";
+        String introduction = logo + H_LINE + introText + H_LINE + "\n";
+        System.out.println(introduction);
+    }
+
+    /* Determine whether the user wants to create a task, exit, list, or mark */
+    private static boolean respondToUser() {
```

Comment:
Good use of naming. Standard words and nouns have been used for naming. Multiple words have been used, but in sensible order for naming